tensorrt_llm/_torch/models/modeling_deepseekv3.py

```diff
@@ -300,11 +300,16 @@ def __init__(
             is_fused=fuse_routing_kernel)
 
     def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
-        # router gemm
-        logits = torch.ops.trtllm.cublas_mm(hidden_states,
-                                            self.weight.t(),
-                                            bias=None,
-                                            out_dtype=torch.float32)
+        if hidden_states.size(0) >= 1 and hidden_states.size(0) <= 16:
+            logits = torch.ops.trtllm.router_gemm_op(hidden_states,
+                                                     self.weight.t(),
+                                                     bias=None,
+                                                     out_dtype=torch.float32)
+        else:
+            logits = torch.ops.trtllm.cublas_mm(hidden_states,
+                                                self.weight.t(),
+                                                bias=None,
+                                                out_dtype=torch.float32)
```

Comment:
Can you move the if condition to inside router_gemm_op? You can call the cublas_mm function inside the torch cpp op.